GitHubTeams.ps1

```diff
@@ -319,131 +319,483 @@ filter Get-GitHubTeamMember
+            -Project 4378613
+
+        Attempts to get a project by its ID for the team 'Everybody' in organization 'PowerShell'.
+        If the team does not have any permissions on the specified project, the API will return an
+        HTTP 404 status code response.
+#>
+    [CmdletBinding(DefaultParameterSetName = 'TeamSlug')]
+    [OutputType({$script:GitHubProjectTypeName})]
+    param
+    (
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName)]
+        [ValidateNotNullOrEmpty()]
+        [String] $OrganizationName,
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName='TeamName')]
+        [ValidateNotNullOrEmpty()]
+        [String] $TeamName,
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName='TeamSlug')]
+        [string] $TeamSlug,
+
+        [Parameter(
+            ValueFromPipelineByPropertyName)]
+        [Alias('ProjectId')]
+        [int64] $Project,
+
+        [string] $AccessToken
+    )
+
+    Write-InvocationLog
+
+    $telemetryProperties = @{
+        OrganizationName = (Get-PiiSafeString -PlainText $OrganizationName)
+        TeamName = (Get-PiiSafeString -PlainText $TeamName)
+        TeamSlug = (Get-PiiSafeString -PlainText $TeamSlug)
+    }
+
+    if ($PSCmdlet.ParameterSetName -eq 'TeamName' -or
+        $PSCmdlet.ParameterSetName -eq 'TeamNameAndProject')
+    {
+        $getGitHubTeamParms = @{
+            OrganizationName = $OrganizationName
+            TeamName = $TeamName
+        }
+
+        if ($PSBoundParameters.ContainsKey('AccessToken'))
+        {
+            $getGitHubTeamParms['AccessToken'] = $AccessToken
+        }
+
+        $team = Get-GitHubTeam @getGitHubTeamParms
+        $TeamSlug = $team.slug
+    }
+
+    if ($Project)
+    {
+        $telemetryProperties['Project'] = Get-PiiSafeString -PlainText $Project
+
+        $params = @{
+            UriFragment = "orgs/$OrganizationName/teams/$TeamSlug/projects/$Project"
+            Method = 'Get'
+            Description = "Getting project $project of team $TeamSlug"
+            AccessToken = $AccessToken
+            TelemetryEventName = $MyInvocation.MyCommand.Name
+            TelemetryProperties = $telemetryProperties
+            AcceptHeader = $script:inertiaAcceptHeader
+        }
+
+        return (Invoke-GHRestMethod @params | Add-GitHubProjectAdditionalProperties)
+    }
+
+    $params = @{
+        UriFragment = "orgs/$OrganizationName/teams/$TeamSlug/projects"
+        Description = "Getting projects of team $TeamSlug"
+        AccessToken = $AccessToken
+        TelemetryEventName = $MyInvocation.MyCommand.Name
+        TelemetryProperties = $telemetryProperties
+        AcceptHeader = $script:inertiaAcceptHeader
+    }
+
+    return (Invoke-GHRestMethodMultipleResult @params | Add-GitHubProjectAdditionalProperties)
+}
+
+filter Get-GitHubTeamRepository
```

Comment:
Same here.  This feels more like functionality that should be in `Get-GitHubRepository` where you could provide the `TeamName` or `TeamSlug` to have it get the repos that are specifically associated with that team.